inc/options/class-wpseo-option-titles.php

```diff
@@ -312,6 +380,86 @@ protected function validate_option( $dirty, $clean, $old ) {
 					}
 					break;
 
+				/* 'post_types-' . $pt->name . '-maintax' fields. */
+				case 'post_types-':
+					$post_type  = str_replace( array( 'post_types-', '-maintax' ), '', $key );
+					$taxonomies = get_object_taxonomies( $post_type, 'names' );
+
+					if ( isset( $dirty[ $key ] ) ) {
+						if ( $taxonomies !== array() && in_array( $dirty[ $key ], $taxonomies, true ) ) {
+							$clean[ $key ] = $dirty[ $key ];
+						}
+						elseif ( (string) $dirty[ $key ] === '0' || (string) $dirty[ $key ] === '' ) {
+							$clean[ $key ] = 0;
+						}
+						elseif ( sanitize_title_with_dashes( $dirty[ $key ] ) === $dirty[ $key ] ) {
+							// Allow taxonomies which may not be registered yet.
+							$clean[ $key ] = $dirty[ $key ];
+						}
+						else {
+							if ( isset( $old[ $key ] ) ) {
+								$clean[ $key ] = sanitize_title_with_dashes( $old[ $key ] );
+							}
+							if ( function_exists( 'add_settings_error' ) ) {
```

Comment:
This `function_exists` check isn't necessary because the function is introduced in WordPress 3.0 and we support 4.8 or 4.9 and higher